server/histogram-aggregator.go

```diff
@@ -0,0 +1,44 @@
+package server
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/codahale/hdrhistogram"
+	"github.com/fluxio/metricd"
+	"github.com/fluxio/metricd/pb"
+)
+
+type histAggregator struct {
+	name   string
+	labels metricd.LabelSet
+	h      *hdrhistogram.Histogram
+}
+
+func (a *histAggregator) AddValue(value float64, ts int64) {
+	a.h.RecordValue(int64(value))
+}
+
+func (a *histAggregator) GetAggregations() []*pb.Metric {
+	if a.h.TotalCount() != 0 {
+		var res []*pb.Metric
+		for _, q := range []float64{0, 50, 90, 95, 99, 100} {
+			res = append(res, &pb.Metric{
+				Name:   a.name + "_p" + strconv.Itoa(int(q)),
+				Labels: a.labels,
+				Value:  &pb.Metric_DoubleValue{float64(a.h.ValueAtQuantile(q))},
+				Ts:     time.Now().Unix(),
+			})
+		}
+		a.h = hdrhistogram.New(0, 1e8, 3)
```

Comment:
Can you call out these values, and why you chose them? Digging in to hdrhistogram, they seem like sane defaults, but users of histAggregator should be able to figure out why it silently (?) fails when they send certain values. I would say that these values should be configurable when creating a histogram aggregator, but I'm fine with encapsulating them until we have a definite need to configure it differently for different metrics. However, we do need to make the limitations obvious to users of metricd.

Also, pull them out so that the constants are in one place.